internal/search/client.go

```diff
@@ -84,18 +88,26 @@ func NewClient(cfg aws.Config, endpoint, indexName string, indexingEnabled bool)
 		return nil, fmt.Errorf("search could not create opensearch client: %w", err)
 	}
 
-	return &Client{indices: svc.Indices, svc: svc, endpoint: endpoint, indexName: indexName, indexingEnabled: indexingEnabled}, nil
+	hash := sha256.New()
+	if err := json.NewEncoder(hash).Encode(indexDefinition); err != nil {
+		return nil, err
+	}
+	sum := hash.Sum(nil)
+
+	return &Client{indices: svc.Indices, svc: svc, endpoint: endpoint, indexName: fmt.Sprintf("%s_%x", indexName, sum[:8]), indexingEnabled: indexingEnabled}, nil
```

Comment:
this adds a hash of the config to the index name, which will prevent problems when it changes but obviously will need the later work to make re-indexing/removal of old ones to be a finished solution